### PR TITLE
fix: Consistent rename behaviour for datasources

### LINKED ISSUE #4904: [Bug] Renaming datasources on the entity explorer forces `_` separator
## Description

If we try to rename a datasource from the datasource editor, naming convention is more lenient than the one allowed via the entity explorer. We should keep the experience consistent on both.

### Steps to reproduce the behaviour:

Add steps to reproduce this behaviour, include console / network logs & screenshots

1. Create a datasource (Eg: Untitled Datasource 1)
2. Navigate to datasource editor and rename it to `Untitled Datasource 2`. This is successful.
3. Now try to rename this datasource from the entity explorer. The name automatically changes to `Untitled_Datasource_2` without performing an edit

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Fixes the rename behavior of datasources in explorer. It will not have the names transformed with underscore anymore and would have a length limit of 30 (just like how it works in the datasources page header).

Fixes #4904 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/4904-datasource-rename-explorer 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.02 **(0.01)** | 36.51 **(0.01)** | 34.73 **(0)** | 55.41 **(0)**
 :red_circle: | app/client/src/pages/Editor/Explorer/Datasources/DatasourceEntity.tsx | 37.5 **(-1.63)** | 0 **(0)** | 0 **(0)** | 37.5 **(-1.63)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/Datasources/DatasourceEntity.tsx [hunk 1]
```diff
@@ -95,6 +95,11 @@ const ExplorerDatasourceEntity = React.memo(
       [datasourceStructure, props.datasource.id, dispatch],
     );
 
+    const nameTransformFn = useCallback(
+      (input: string) => input.slice(0, 30),
```
Comment (akash-codemonk) [inline app/client/src/pages/Editor/Explorer/Datasources/DatasourceEntity.tsx:99]: Not sure does it work well with previous datasource names. Will they get overwritten when we try to edit a name more that 30 characters and whether that would be a problem to existing users?

**Review by akash-codemonk — APPROVED** [review]

File: app/client/src/pages/Editor/Explorer/Datasources/DatasourceEntity.tsx (hunk 1 above)
Comment (arunvjn) [reply app/client/src/pages/Editor/Explorer/Datasources/DatasourceEntity.tsx:99]: Datasource names has always had 30 characters as the limit. So shouldn't be a problem

**Pull request merged**